octane/evmengine/keeper/db.go

```diff
@@ -160,5 +161,28 @@ func (k *Keeper) EligibleWithdrawals(ctx context.Context) ([]*Withdrawal, error)
 		}
 	}
 
-	return withdrawals, nil
+	evmWithdrawals := []*etypes.Withdrawal{}
+	for _, w := range withdrawals {
+		evmWithdrawals = append(evmWithdrawals, &etypes.Withdrawal{
+			Index:   w.GetId(),
+			Address: common.BytesToAddress(w.GetAddress()), //nolint:forbidigo // should be padded
+			Amount:  w.GetAmountGwei(),
+			// The validator index is not used for withdrawals.
+			Validator: 0,
+		})
+	}
+
+	return evmWithdrawals, nil
+}
+
+// RemoveWithdrawals removes all passed withdrawals by the id.
+func (k *Keeper) RemoveWithdrawals(ctx context.Context, withdrawals []*etypes.Withdrawal) error {
```

Comment:
same here, don't export by default

Reply:
nit: maybe use a more "DB" term "deleteWithdrawals"

Reply:
we also need a unit test for this function (ensure it doesn't delete other withdrawals).

Reply:
added